### PR TITLE
feat: add admin modal component

### PR DESCRIPTION
**Pull request by chenjanet**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Add-Admin-Modal-Component-f4363bbc18b8448aadf0ead8b5271252)

## Implementation description
* `components/common/AddAdminModal.tsx`: create new `AddAdminModal` to support adding new admins
* `components/common/icons.tsx`: added `PlusOutlineIcon` (+)
* `components/pages/ComponentLibrary.tsx`: added new component to component library

## What should reviewers focus on?
Screenshots:
Screen Shot 2022-10-30 at 2 13 30 PM
Screen Shot 2022-10-30 at 2 13 55 PM
Screen Shot 2022-10-30 at 2 14 07 PM

Figma: https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=484%3A17341

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/common/AddAdminModal.tsx [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        <ModalContent p={2}>
+          <ModalHeader>
+            <Text textStyle="subtitle1" color="grey.400">
+              Add Admin
+            </Text>
+          </ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <FormControl isRequired>
+              <FormLabel color="blue.300" fontSize="20px" lineHeight="26px">
```
Comment (chenjanet) [inline frontend/src/components/common/AddAdminModal.tsx]: Tried using `textStyle` here but it wouldn't apply, so these are hard-coded. This seems to be a known issue, according to this issue in the Chakra UI repo: https://github.com/chakra-ui/chakra-ui/issues/4566

File: frontend/src/components/common/AddAdminModal.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                as="legend"
+                textStyle="subtitle2"
+                color="blue.300"
+                mb={0}
+                fontSize="20px"
+                lineHeight="26px"
+              >
+                Does the user already have a Jump Math email address?
+              </FormLabel>
+              <RadioGroup
```
Comment (joyce-shi) [inline frontend/src/components/common/AddAdminModal.tsx]: nit: but can we update the radio buttons to use the background colour specified by design? I think its the `blue.300`

File: frontend/src/components/common/AddAdminModal.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    console.log("creating admin with data");
+    console.log(
+      `first name: ${firstName}\nlast name: ${lastName}\nemail: ${email}`,
+    );
+    onModalClose();
+  };
+
+  return (
+    <>
+      <Button
```
Comment (joyce-shi) [inline frontend/src/components/common/AddAdminModal.tsx:60]: Is this the button to `Add Admin` from the parent page? If so, can we separate this out into a different component and then leave the logic to open this modal when it's clicked?

File: frontend/src/components/common/AddAdminModal.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                    Yes
+                  </Radio>
+                </HStack>
+              </RadioGroup>
+            </FormControl>
+            {hasJumpMathEmail !== null && (
+              <>
+                <FormControl isRequired mt={6}>
+                  <AddAdminFormLabel
+                    text={`Please enter their ${
```
Comment (joyce-shi) [inline frontend/src/components/common/AddAdminModal.tsx]: In the case that the user selects that they have a Jump Math email address can we validate that they use that email address? Also, if we could pre-fill the field with the Jump Math email address format (i.e `@jumpmath.org`)

File: frontend/src/components/common/AddAdminModal.tsx [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+                  <AddAdminTextInput
+                    validationType="email"
+                    placeholder="Email Address"
+                    handleChange={setConfirmEmail}
+                  />
+                </FormControl>
+              </>
+            )}
+          </ModalBody>
+          <ModalFooter>
```
Comment (joyce-shi) [inline frontend/src/components/common/AddAdminModal.tsx]: maybe we can separate this out into a separate component? We can probably use this at another point in time anyway.

File: frontend/src/components/common/AddAdminModal.tsx [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+  );
+
+  const onModalClose = () => {
+    setHasJumpMathEmail(null);
+    onClose();
+  };
+
+  const onAddAdminClick = () => {
+    if (email !== confirmEmail) {
+      console.log("emails don't match");
```
Comment (joyce-shi) [inline frontend/src/components/common/AddAdminModal.tsx]: Are we supposed to show an error state here? If so, can we make sure that there is another ticket to add this functionality!

File: frontend/src/components/common/AddAdminModal.tsx [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+      mb={0}
+      fontSize="20px"
+      lineHeight="26px"
+    >
+      {text}
+    </FormLabel>
+  );
+};
+
+const AddAdminTextInput = ({
```
Comment (joyce-shi) [inline frontend/src/components/common/AddAdminModal.tsx]: Maybe we can actually refactor `AddAdminTextInput` and `AddAdminFormLabel` to be a general components for our codebase when creating forms? Let me know if there's some reason why this shouldn't be the case!

**Review by joyce-shi — CHANGES_REQUESTED** [review]
Overall LGTM 🔥 One thing: IMO, the `AddAdminModal` component is a bit lengthy for a component. I left a couple comments about refactoring & separating some things out, let me know your thoughts!

In general, if there are more things that could be refactored out into constants, maybe that would help a bit with readability and bulk!

File: frontend/src/components/common/AddAdminModal.tsx (hunk 3 above)
Comment (chenjanet) [reply frontend/src/components/common/AddAdminModal.tsx:60]: Not sure if that's possible, given the way that the `useDisclosure()` hook is used in the examples in the Chakra UI documentation (https://chakra-ui.com/docs/components/modal) :/

Comment (joyce-shi) [reply frontend/src/components/common/AddAdminModal.tsx:60]: oops my bad! thanks for letting me know (:

**Review by joyce-shi — APPROVED** [review]
lgtm!

**Comment by vyomeshiyengar** [comment]
Is there a state implemented when the admin clicks save without the required fields populated?

**Comment by chenjanet** [comment]
> Is there a state implemented when the admin clicks save without the required fields populated?

Yes! Although it's just the native HTML form validation (+ Chakra UI's wrapper around it possibly)? Here are a few screenshots:

Screen Shot 2022-11-14 at 6 02 12 PM

Screen Shot 2022-11-14 at 6 03 08 PM

**Comment by vyomeshiyengar** [comment]
Would it be possible to add this error state -
 https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=853%3A35439
<https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=853%3A35439>?
If not all good

**Pull request merged**